lib/ret_web/controllers/api/v1/oauth_controller.ex

```diff
@@ -29,7 +35,31 @@ defmodule RetWeb.Api.V1.OAuthController do
     end
   end
 
-  def show(conn, %{"type" => "discord", "error" => "access_denied", "state" => state}) do
+  # handle the chat app oauth information: discord, slack etc.
+  def show(conn, %{"type" => _type} = params) do
```

Comment:
You could make this code more explicit and self-documenting by using a `when` guard:
```elixir
def show(conn, %{"type" => type} = params) when type in ["discord", "slack"] do
  handle_chat_oauth(params, conn)
end
```